src/collapse/collapse-content.tsx

```diff
@@ -36,19 +35,36 @@ export const CollapseContent: React.FC<PropsWithChildren<Props>> = ({
   const contentRef = useRef<HTMLDivElement | null>(null);
   const initialContentHeight = useRef<number>(minHeight);
   const contentHeight = useRef<number>(DEFAULT_HEIGHT);
-  const [dimensions, setDimensions] = useState({
-    width: 0,
-    height: 0,
-  });
+
+  const [dimensions, setDimensions] = useState({width: 0, height: 0});
   const [height, setHeight] = useState<string>(toPx(minHeight));
-  const [showFade, setShowFade] = useState<boolean>(true);
+  const [showFade, setShowFade] = useState<boolean>(collapsed);
+
+  const [shouldHideContent, setShouldHideContent] = useState<boolean>(
+    collapsed && initialContentHeight.current <= DEFAULT_HEIGHT,
+  );
+
+  const onTransitionEnd = useCallback(() => {
+    if (initialContentHeight.current <= DEFAULT_HEIGHT) {
+      setShouldHideContent(collapsed);
+    }
+  }, [collapsed]);
 
   useEffect(() => {
-    if (!collapsed) {
-      setShowFade(false);
-    } else {
-      setShowFade(true);
+    const ref = containerRef.current;
+    if (ref) {
+      ref.addEventListener('transitionend', onTransitionEnd);
     }
+    return () => {
+      if (ref) {
+        ref.removeEventListener('transitionend', onTransitionEnd);
+      }
+    };
+  }, [onTransitionEnd]);
```

Comment:
We can move `onTransitionEnd` inside the effect, and then we don't need `useCallback` (https://react.dev/learn/removing-effect-dependencies#move-dynamic-objects-and-functions-inside-your-effect):

```ts
  useEffect(() => {
    function onTransitionEnd() {
      if (initialContentHeight.current <= DEFAULT_HEIGHT) {
        setShouldHideContent(collapsed);
      }
    }

    const container = containerRef.current;
    if (container) {
      container.addEventListener('transitionend', onTransitionEnd);
    }
    return () => {
      if (container) {
        container.removeEventListener('transitionend', onTransitionEnd);
      }
    };
  }, [collapsed]);
```

Reply:
@vkrol Thanks for the suggestion 👍
Done ✅